test/helpers/position_assertions.cc

```diff
@@ -1233,6 +1239,112 @@ string ApplyCompletionAssertion::toString() const {
+        fmt::format("Improperly formatted apply-rename assertion. Expected '[<version>] <index>'. Found '{}'",
+                    assertionContents));
+
+    return nullptr;
+}
+
+ApplyRenameAssertion::ApplyRenameAssertion(string_view filename, unique_ptr<Range> &range, int assertionLine,
+                                           string_view version, string newName)
+    : RangeAssertion(filename, range, assertionLine), version(string(version)), newName(newName) {}
+
+void ApplyRenameAssertion::checkAll(const vector<shared_ptr<RangeAssertion>> &assertions,
+                                    const UnorderedMap<string, shared_ptr<core::File>> &sourceFileContents,
+                                    LSPWrapper &wrapper, int &nextId, string errorPrefix) {
+    for (auto assertion : assertions) {
+        if (auto assertionOfType = dynamic_pointer_cast<ApplyRenameAssertion>(assertion)) {
+            assertionOfType->check(sourceFileContents, wrapper, nextId, errorPrefix);
+        }
+    }
+}
+
+void ApplyRenameAssertion::check(const UnorderedMap<std::string, std::shared_ptr<core::File>> &sourceFileContents,
+                                 LSPWrapper &wrapper, int &nextId, std::string errorPrefix) {
+    REQUIRE_FALSE(this->newName.empty());
+
+    auto renameList = doTextDocumentRename(wrapper, *this->range, nextId, this->filename, newName);
+    {
+        INFO("doTextDocumentRename failed; see error above.");
+        REQUIRE_NE(renameList, nullptr);
+    }
+
+    auto &renameItems = *renameList->documentChanges;
+    auto it = sourceFileContents.find(this->filename);
+    {
+        INFO(fmt::format("Unable to find source file `{}`", this->filename));
+        REQUIRE_NE(it, sourceFileContents.end());
+    }
+
+    size_t index = this->filename.rfind("/", this->filename.length());
+    auto testDataPath = this->filename.substr(0, index);
+    UnorderedMap<string, string> sourceFiles;
+    // Build a map of all the .rb and .rbi files
+    for (auto filePath : FileOps::listFilesInDir(testDataPath, {".rb", ".rbi"}, false, {}, {})) {
+        sourceFiles[filePath] = FileOps::read(filePath);
+    }
+
+    string sourceFilePath;
+    string expectedEditedFilePath;
+    string expectedEditedFileContents;
+    string actualEditedFileContents;
+    for (auto &renameItem : renameItems) {
+        // Get the file path from the edited document's uri so we can determine the path to the .rbedited file
+        index = renameItem->textDocument->uri.find(testDataPath, 0);
+        sourceFilePath = renameItem->textDocument->uri.substr(index, renameItem->textDocument->uri.length() - 1);
+        expectedEditedFilePath = updatedFilePath(sourceFilePath, this->version);
+
+        auto &edits = renameItem->edits;
+        try {
+            expectedEditedFileContents = FileOps::read(expectedEditedFilePath);
+        } catch (FileNotFoundException e) {
+            ADD_FAIL_CHECK_AT(filename.c_str(), this->assertionLine + 1,
```

Comment:
I think you also need to check that all files which have a `this->version` version have been edited.

e.g., if you have `test__foo.rb`, `test__foo.A.rbedited`, `test__bar.rb`, and `test__bar.A.rbedited`, and rename only renames `foo`, the test should fail.